staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -371,14 +387,16 @@ Loop:
 }
 
 // parseQuote unquotes string inside double quote
-func (p *Parser) parseQuote(cur *ListNode) error {
+func (p *Parser) parseQuote(cur *ListNode, end rune) error {
```

Comment:
Like you said, this is using strconv.Unquote, which is for unquoting Go language constructs. jsonpath is not Go, and 'blah' in jsonpath is a string, not a rune. We may have to roll our own Unquote.